### ISSUE TITLE
Unable to change station deployment receiver - also cannot delete the deployment

### ISSUE THREAD

**Issue by eageldart** [MotusDev/Motus-TO-DO#1123]
I am trying to change the Sensorgnome serial # for receiver deployment 8434 of Project 78. When I try to do so using the 'Update receiver properties' button, I get an error associated with the following reference number: 1646754236909. I have attached some screenshots of the error message. 

![Error reference number 1646754236909 A](https://user-images.githubusercontent.com/94930170/157274664-fc6b7830-f9e1-46e0-bb55-1b851e0f40ed.png)
![Error reference number 1646754236909 B](https://user-images.githubusercontent.com/94930170/157274665-3a26f7e6-ec97-42a2-9d38-93c1d1e42496.png)
![Error reference number 1646754236909 C](https://user-images.githubusercontent.com/94930170/157274666-26fc879d-a3f3-4037-b9bd-598bbdc42849.png)

I have also tried to delete this deployment to make a new one with the proper Sensorgnome associated with it. When I click 'Delete deployment' and press 'OK', the deployment remains in the Deployment History table. I have only had this problem with this particular deployment - I have been able to delete others.

**Comment by leberrigan** [comment]
Can you take a look at that example again and post links to the deployment and provide the receiver ID and serial number? Just trying to keep this from getting buried

**Comment by eageldart** [comment]
Here is where I am trying to Update receiver properties and Delete deployment: 
https://motus.org/data/station/deployment/edit?projectID=78&id=8434

I want to change the Receiver ID from SG-0B2DRPI3DBAC (current, incorrect) to SG-0855RPI39CDE (correct)

I checked and the receiver that I want to switch it to was terminated.

**Comment by cjardine-bsc** [comment]
The problem is a duplicate in the DB. 

This ReceiverID is already registered to this project

You need to move the deployment to the correct receiver and then I can delete the bad serial number. 

Something for the website list, the webpage should handle the response from the proc that the receiver is a duplicate and display an error message to the user.

**Comment by leberrigan** [comment]
The issue here is that we can't change the `ReceiverID` for this deployment to the correct one because it isn't showing up in the list of 'undeployed receivers', even though it isn't deployed. We also can't delete the deployment, as Erica described. 

To work around this issue, I modified the erroneous deployment so it ends 1 minute after starting (on 2021-07-20) and then added a new deployment to replace it (with correct dates). In this new deployment I was able to select the correct receiver.

We still can't delete the erroneous deployment, though.

@cjardine-bsc, can you delete deployment 8434 for us? https://motus.org/data/station/deployment/edit?projectID=78&id=8434

Next step is to document this in another Git issue so we can work towards a fix.

**Comment by cjardine-bsc** [comment]
Ahhh okay, the problem is the dropdown list. The receiver you are looking for is not there. Now I understand. 

![image](https://user-images.githubusercontent.com/22172758/158458713-26202e2a-a855-4a83-8f3d-c48bee9ae7bd.png)

You're trying to move a deployment from 2021-07-20 to that receiver. 

![image](https://user-images.githubusercontent.com/22172758/158459018-34cd693d-fa6c-446a-bba8-7e2175978098.png)

BUT that receiver has a deployment that overlaps with that deployment. 

![image](https://user-images.githubusercontent.com/22172758/158459339-324ef7c5-cd74-4ee1-924a-17a59b1b75cd.png)


Change the deployment dates so they don't overlap and I expect that the receiver will show up in the list. 

Give that a test, if it doesn't work, then the logic on the drop down is wrong and we can bring in J-dawg for a fix.

**Comment by cjardine-bsc** [comment]
If speed is an issue let me know and I can manually update the sensor_id for the deployment.

**Comment by eageldart** [comment]
I just successfully deleted deployment 8434. I am going to close this issue now.

**Comment by cjardine-bsc** [comment]
@eageldart  How? I would like to confirm that there is not a problem with the drop down list before closing this.

**Comment by leberrigan** [comment]
I think there is still a lingering issue, we just may have lost our example so keep an eye out for another one.

There were two separate issues here:
1. There _wasn't_ another deployment for that receiver and I still couldn't select it. However, I could select it if I created a new deployment rather than editing the old one.
2. The second problem was that we weren't able to delete that deployment (8434) for some reason. And then something happened after I made the edits [as mention above](https://github.com/MotusDev/Motus-TO-DO/issues/1123#issuecomment-1065026083) that made it possible to delete that deployments. However, I don't know if was something I had done because I still wasn't able to delete the deployment - Erica ended up doing it a few days later.
